include/HtmlSanitizer.php

```diff
@@ -0,0 +1,142 @@
+    {
+        $configurator = new \Configurator();
+        $sugar_config = $configurator->config;
+
+
+        $config = \HTMLPurifier_Config::createDefault();
+
+        if (!is_dir(sugar_cached("htmlclean"))) {
+            create_cache_directory("htmlclean/");
+        }
+
+        $config->set('HTML.Doctype', 'XHTML 1.0 Transitional');
+        $config->set('Core.Encoding', 'UTF-8');
+        $hidden_tags = array('script' => true, 'style' => true, 'title' => true, 'head' => true);
+        $config->set('Core.HiddenElements', $hidden_tags);
+        $config->set('Cache.SerializerPath', sugar_cached("htmlclean"));
+        $config->set('URI.Base', isset($sugar_config['site_url']) ? $sugar_config['site_url'] : null);
+        $config->set('CSS.Proprietary', true);
+        $config->set('HTML.TidyLevel', 'light');
+        $config->set('HTML.ForbiddenElements', array('body' => true, 'html' => true));
+        $config->set('AutoFormat.RemoveEmpty', true);
+        $config->set('Cache.SerializerPermissions', 0775);
+        $config->set('Filter.ExtractStyleBlocks.TidyImpl', false);
+        if (!empty($sugar_config['html_allow_objects'])) {
+            $config->set('HTML.SafeObject', true);
+            $config->set('HTML.SafeEmbed', true);
+        }
+        $config->set('Output.FlashCompat', true);
+        $config->set('Filter.Custom', array(new HTMLPurifierFilterXmp()));
+        $config->set('HTML.DefinitionID', 'Sugar HTML Def');
+        $config->set('HTML.DefinitionRev', 2);
+        $config->set('Cache.SerializerPath', sugar_cached('htmlclean/'));
+        $config->set('Attr.EnableID', true);
+        $config->set('Attr.IDPrefix', 'sugar_text_');
+
+        if ($def = $config->maybeGetRawHTMLDefinition()) {
+            $iframe = $def->addElement(
+                'iframe',
+                'Flow',
+                'Optional: #PCDATA | Flow | Block',
+                'Core',
+                array(
+                    'src*' => 'URI',
+                    'frameborder' => 'Enum#0,1',
+                    'marginwidth' => 'Pixels',
+                    'marginheight' => 'Pixels',
+                    'scrolling' => 'Enum#|yes,no,auto',
+                    'align' => 'Enum#top,middle,bottom,left,right,center',
+                    'height' => 'Length',
+                    'width' => 'Length',
+                )
+            );
+
+            $iframe->excludes = array('iframe');
+        }
+
+        /** @var \HTMLPurifier_URIDefinition $uri */
+        $uri = $config->getDefinition('URI');
+        $uri->addFilter(new URIFilter(), $config);
+        \HTMLPurifier_URISchemeRegistry::instance()->register('cid', new HTMLPurifierURISchemeCid());
+
+        $this->purifier = new \HTMLPurifier($config);
+    }
+
+    /**
+     * Get cleaner instance
+     * @return HtmlSanitizer
+     */
+    public static function getInstance()
+    {
+        return self::$instance instanceof self ? self::$instance : (self::$instance = new self());
```

Comment:
if (!self::$instance) {
// populate
}

return self::$instance;

Reply:
Yes - getInstance is a singleton pattern, I will change this correctly match this pattern. :D